scalding-core/src/main/scala/com/twitter/scalding/TimePathedSource.scala

```diff
@@ -31,6 +31,38 @@ object TimePathedSource {
       "%1$tm" -> Months(1)(tz), "%1$tY" -> Years(1)(tz))
       .find { unitDur: (String, Duration) => pattern.contains(unitDur._1) }
       .map(_._2)
+
+  /**
+   * Gives all paths in the given daterange with windows based on the provided duration.
+   */
+  def allPathsWithDuration(pattern: String, duration: Option[Duration], dateRange: DateRange, tz: TimeZone): Iterable[String] =
+    duration.map { dur =>
+      // This method is exhaustive, but too expensive for Cascading's JobConf writing.
+      dateRange.each(dur)
+        .map { dr: DateRange =>
+          this.toPath(pattern, dr.start, tz)
+        }
+    }.getOrElse(Nil)
+
+  /**
+   * Gives all read paths in the given daterange.
+   */
+  def readPathsFor(pattern: String, dateRange: DateRange, tz: TimeZone): Iterable[String] = {
+    val stepSize = TimePathedSource.stepSize(pattern, DateOps.UTC)
+    this.allPathsWithDuration(pattern, stepSize, dateRange, DateOps.UTC)
+  }
+
+  /**
+   * Gives the write path based on daterange end.
+   */
+  def writePathFor(pattern: String, dateRange: DateRange, tz: TimeZone): String = {
+    // TODO this should be required everywhere but works on read without it
+    // maybe in 0.9.0 be more strict
```

Comment:
Seems to me like this is actually needed to globify read paths but has to be removed for the write path?